context/src/main/java/io/grpc/Context.java

```diff
@@ -49,30 +49,33 @@
  *   <li>Local and distributed tracing information.</li>
  * </ul>
  *
- * <p>Context objects make their state available by being attached to the executing thread using
- * a {@link ThreadLocal}. The context object bound to a thread is considered {@link #current()}.
- * Context objects are immutable and inherit state from their parent. To add or overwrite the
- * current state a new context object must be created and then attached to the thread replacing the
- * previously bound context. For example:
+ * <p>A Context object can be {@link #attach attached} to the {@link Storage}, which effectively
+ * forms a <b>scope</b> for the context.  The scope is typically bound to the current thread, though
+ * alternative {@link Storage} implementations can alter this behavior.  Within a scope, its Context
```

Comment:
I think Context becomes useless if it isn't bound in some way to the current thread. The scope has no meaning then, since it is basically undefined. All the use cases we're considering now are all thread-local-like storage.